src/utils/poolData/kyber.ts

```diff
@@ -1,41 +1,63 @@
 import { BigNumber, Contract } from 'ethers'
 
 import {
+  ChainId,
   KYBER_FACTORY,
   KYBER_FACTORY_ABI,
+  KYBER_FACTORY_POLYGON,
   KYBER_POOL_ABI,
   TEN_POW_18,
 } from 'utils/constants/constants'
-import { WETH } from 'utils/constants/tokens'
+
 import { getProvider } from 'utils/provider'
+import { getWETH } from 'utils/weth'
 
 type KyberBalances = {
   tokenBalance: BigNumber
   wethBalance: BigNumber
 }
 
+function getFactoryAddress(chainId: ChainId) {
+  switch (chainId) {
+    case ChainId.ethereum:
+      return KYBER_FACTORY
+    case ChainId.polygon:
+      return KYBER_FACTORY_POLYGON
+  }
+}
+
 export async function getKyberLiquidity(
-  tokenAddress: string
+  tokenAddress: string,
+  chainId: ChainId
 ): Promise<KyberBalances> {
   let response: KyberBalances = {
     tokenBalance: BigNumber.from(0),
     wethBalance: BigNumber.from(0),
   }
-  const provider = getProvider()
-  const factoryInstance = await new Contract(
-    KYBER_FACTORY,
-    KYBER_FACTORY_ABI,
-    provider
-  )
 
-  const pools = await factoryInstance.getPools(tokenAddress, WETH)
-  if (!pools[0]) return response
+  try {
+    const provider = getProvider(chainId)
+    const WETH = getWETH(chainId)
+    const factoryAddress = getFactoryAddress(chainId)
+    const factoryInstance = await new Contract(
+      factoryAddress,
+      KYBER_FACTORY_ABI,
+      provider
+    )
+
+    const pools = await factoryInstance.getPools(tokenAddress, WETH)
+    if (!pools[0]) return response
 
-  const pairContract = await new Contract(pools[0], KYBER_POOL_ABI, provider)
-  const [tokenBalance, wethBalance] = await pairContract.getReserves()
+    const pairContract = await new Contract(pools[0], KYBER_POOL_ABI, provider)
+    const [tokenBalance, wethBalance] = await pairContract.getReserves()
 
-  response.tokenBalance = tokenBalance.div(TEN_POW_18)
-  response.wethBalance = tokenBalance.div(wethBalance)
+    response.tokenBalance = tokenBalance.div(TEN_POW_18)
+    response.wethBalance = tokenBalance.div(wethBalance)
 
-  return response
+    return response
+  } catch (error) {
+    console.log('Error getting liquidity from kyber')
+    console.log(error)
+    return response
+  }
```

Comment:
Just found Kyber has a [SDK](https://github.com/dynamic-amm/dmm-sdk/tree/main/src). Might be worth a try for a different approach?